base_layer/core/src/transaction.rs

```diff
@@ -503,8 +500,14 @@ impl Transaction {
         &self.body
     }
 
-    pub fn get_weight(&self) -> Result<usize, MessageFormatError> {
-        Ok(self.to_binary()?.len())
+    /// Returns the byte size or weight of a transaction
+    pub fn calculate_weight(&self) -> u64 {
+        Fee::calculate_weight(self.body.inputs.len(), self.body.outputs.len())
+    }
+
+    /// Returns the total fee allocated to each byte of the transaction
+    pub fn calculate_ave_fee_per_gram(&self) -> Result<f64, MessageFormatError> {
```

Comment:
How can this return an Error?

Shouldn't it just return f64?

Reply:
Good catch, change applied